icu4c/source/common/uloc_tag.cpp

```diff
@@ -67,6 +66,9 @@ typedef struct ULanguageTag {
 #define ISALPHA(c) uprv_isASCIILetter(c)
 #define ISNUMERIC(c) ((c)>='0' && (c)<='9')
 
+static const char SEP[] = { '-' };
```

Comment:
Why character arrays for single-character constants? Seems misleading/unnecessary/confusing.

And why not just use those single characters directly? (I realize that the old code already used macros.)

Reply:
This was just an attempt to keep the new code similar to the old code, but maybe a better way forward is to instead from here on start moving towards using string literals instead of these macros. Changed.